x-pack/plugins/security_solution/server/lib/entity_analytics/utils/create_or_update_index.ts

```diff
@@ -51,7 +55,24 @@ export const createOrUpdateIndex = async ({
         );
       }
     } else {
-      return esClient.indices.create(options);
+      try {
+        await esClient.indices.create(options);
+        if (waitForStatus) {
+          await waitForIndexStatus({
+            index: options.index,
+            status: waitForStatus,
+            esClient,
+            logger,
+          });
+        }
+      } catch (err) {
+        // If the index already exists, we can ignore the error
+        if (err?.meta?.body?.error?.type === 'resource_already_exists_exception') {
```

Comment:
Should this be the approach going forward, where when trying to initialise resources, if it's already there, we swallow the error? 
This applies mostly to non entity store resources, because we usually delete all entity engine related resources on init.

The other solution would be to have some sort of lock mechanism, but that seems hard 😅

Reply:
We would have to take it resource by resource I think, I agree locking is not going to be do-able, I prefer this optimistic approach.